### PR TITLE
refactoring `default` and `on update` expressions

### PR DESCRIPTION
**Pull request by jycor**
This PR changes the grammar to more closely match MySQL's behavior, specifically around the `NOW()` function and its synonyms. 
Changes:
 - Throw syntax errors for `ON UPDATE` expressions against functinos that aren't `NOW()` or a synonym.
 - Only accept integer for argument to `NOW()` and synonyms; syntax error for anything else
 - Simplified grammar rules
 - Removed `CurTimeFuncExpr` from AST in favor of plain `FuncExpr`

Companion PR: https://github.com/dolthub/go-mysql-server/pull/2218

### CODE REVIEW

File: go/vt/sqlparser/sql.y [hunk 1]
```diff
@@ -7096,98 +7068,89 @@ function_call_keyword:
...
+
 // Optional parens for certain keyword functions that don't require them.
 func_parens_opt:
   /* empty */
 | openb closeb
 
-func_datetime_precision:
-  openb value_expression closeb
+// Optional datetime precision for certain functions.
+func_datetime_prec_opt:
```
Comment (max-hoffman) [inline go/vt/sqlparser/sql.y:7145]: do we have error tests with non-integer precisions?

File: go/vt/sqlparser/sql.y [hunk 2]
```diff
@@ -7096,98 +7068,89 @@ function_call_keyword:
...
+    $$ = &FuncExpr{Name: NewColIdent(string($1)), Exprs: SelectExprs{&AliasedExpr{Expr: $2}}}
   }
-| EXTRACT openb time_unit FROM value_expression closeb
+| LOCALTIMESTAMP func_datetime_prec_opt
   {
-    $$ = &ExtractFuncExpr{Name: string($1), Unit: string($3), Expr: $5}
+    $$ = &FuncExpr{Name: NewColIdent(string($1)), Exprs: SelectExprs{&AliasedExpr{Expr: $2}}}
   }
 
+
```
Comment (max-hoffman) [inline go/vt/sqlparser/sql.y:7138]:
```suggestion
```

**Review by max-hoffman — APPROVED** [review]
lgtm, the specifics of the diff are a bit hard to understand but the api refactor makes sense

File: go/vt/sqlparser/sql.y (hunk 1 above)
Comment (jycor) [reply go/vt/sqlparser/sql.y:7145]: Yes, they are here:
https://github.com/dolthub/vitess/pull/296/files#diff-f888604778470e9f0188a9034ce01275a29c0bc35a2edc76b3b08ae279ef9f69R4518-R4549

**Pull request merged**